core/src/test/java/com/yahoo/oak/SingleThreadIteratorTest.java

```diff
@@ -485,4 +485,52 @@ public void testDescending() {
 
     }
 
+    @Test
+    public void testIteratorRemove() {
```

Comment:
Can you please add some corner cases for testing? Like when the previous state is on the other chunk, the key to be deleted via iterator is also deleted else how, parallel update, etc.

Reply:
Only things related to single thread here and multi-threading related corner cases in other test

Reply:
Is there a check that entries number is getting less?

Reply:
1. With such small number of entries, it's almost sure the previous thread is in another chunk.
2. The HashMap is currently always single threaded, isn't it?
3. we check all the entries present and make sure we have only the correct entries after the remove